packages/lib-classifier/src/store/feedback/strategies/datavis/graph2drange/reducer.spec.js

```diff
@@ -2,89 +2,164 @@ import { expect } from 'chai'
+        x: '15'
+      })
     })
-  })
 
-  it('should return result with success', function () {
-    expect(graph2dRangeReducer(rule, [annotationSuccess])).to.deep.equal({
-      hideSubjectViewer: false,
-      id: '1234',
-      strategy: 'graph2dRange',
-      success: true,
-      successEnabled: true,
-      successMessage: 'Success!',
-      successfulClassifications: [annotationSuccess],
-      tolerance: '1',
-      width: '2',
-      x: '15'
+    it('should return result with success', function () {
+      expect(graph2dRangeReducer(ruleWithStrings, [annotationSuccess])).to.deep.equal({
+        hideSubjectViewer: false,
+        id: '1234',
+        strategy: 'graph2dRange',
+        success: true,
+        successEnabled: true,
+        successMessage: 'Success!',
+        successfulClassifications: [annotationSuccess],
+        tolerance: '0.25',
+        width: '0.1',
+        x: '15'
+      })
+    })
+
+    it('should return result within tolerance', function () {
+      expect(graph2dRangeReducer(ruleWithStrings, [annotationTolerance])).to.deep.equal({
+        hideSubjectViewer: false,
+        id: '1234',
+        strategy: 'graph2dRange',
+        success: true,
+        successEnabled: true,
+        successMessage: 'Success!',
+        successfulClassifications: [annotationTolerance],
+        tolerance: '0.25',
+        width: '0.1',
+        x: '15'
+      })
     })
-  })
 
-  it('should return result within tolerance', function () {
-    expect(graph2dRangeReducer(rule, [annotationTolerance])).to.deep.equal({
-      hideSubjectViewer: false,
-      id: '1234',
-      strategy: 'graph2dRange',
-      success: true,
-      successEnabled: true,
-      successMessage: 'Success!',
-      successfulClassifications: [annotationTolerance],
-      tolerance: '1',
-      width: '2',
-      x: '15'
+    it('should return result with success with successful and failed annotation', function () {
+      expect(graph2dRangeReducer(ruleWithStrings, [annotationSuccess, annotationTolerance, annotationFailure])).to.deep.equal({
+        hideSubjectViewer: false,
+        id: '1234',
+        strategy: 'graph2dRange',
+        success: true,
+        successEnabled: true,
+        successMessage: 'Success!',
+        successfulClassifications: [annotationSuccess, annotationTolerance],
+        tolerance: '0.25',
+        width: '0.1',
+        x: '15'
+      })
     })
   })
 
-  it('should return result with success with successful and failed annotation', function () {
-    expect(graph2dRangeReducer(rule, [annotationSuccess, annotationTolerance, annotationFailure])).to.deep.equal({
-      hideSubjectViewer: false,
-      id: '1234',
-      strategy: 'graph2dRange',
-      success: true,
-      successEnabled: true,
-      successMessage: 'Success!',
-      successfulClassifications: [annotationSuccess, annotationTolerance],
-      tolerance: '1',
-      width: '2',
-      x: '15'
+  describe('with rule data as numbers', function () {
```

Comment:
There's a lot of repeated code from the previous block in this function, which could be moved into a shared function.
```javascript
describe('with rule data as strings', function () {
 testFeedbackRule(ruleWithStrings);
}
describe('with rule data as numbers', function () {
 testFeedbackRule(ruleWithNums);
}
```
where `testFeedbackRule` is a function that takes a mock feedback rule and passes it to a series of `it()` functions.